src/main/java/gregtech/api/capability/impl/EnergyContainerBatteryBuffer.java

```diff
@@ -111,31 +134,49 @@ public void update() {
                 return;
             }
 
+            if (getOutputAmperage() == 0)
+                return;
+
             IItemHandlerModifiable inventory = getInventory();
-            long voltage = getOutputVoltage();
-            long maxAmperage = 0L;
-            TIntList slotsList = new TIntArrayList();
+            // distribute energy evenly
+            long distributed = getOutputVoltage();
+            if (distributed == 0)
+                return;
+
             for (int i = 0; i < inventory.getSlots(); i++) {
                 ItemStack batteryStack = inventory.getStackInSlot(i);
                 IElectricItem electricItem = getBatteryContainer(batteryStack);
-                if (electricItem == null) continue;
-                if (electricItem.discharge(voltage, getTier(), true, true, true) == voltage) {
-                    slotsList.add(i);
-                    maxAmperage++;
+                // non batteries and empty batteries are ignored
+                if (electricItem == null || electricItem.getCharge() == 0)
+                    continue;
+
+                // if the potential distributed voltage cannot all drain,
+                // decrease distributed voltage so it will not overdrain the battery
+                if (electricItem.discharge(distributed, getTier(), true, true, true) != distributed) {
+                    long space = distributed - electricItem.getCharge();
+                    distributed -= space;
                 }
             }
-            if (maxAmperage == 0) return;
-            long amperageUsed = energyContainer.acceptEnergyFromNetwork(outFacing.getOpposite(), voltage, maxAmperage);
-            if (amperageUsed == 0) return;
-            energyOutputPerSec += amperageUsed * voltage;
-            for (int i : slotsList.toArray()) {
+            // no energy can be distributed, so do nothing
+            if (distributed <= 0)
+                return;
+
+            long amperageUsed = energyContainer.acceptEnergyFromNetwork(outFacing.getOpposite(), distributed / getOutputAmperage(), getOutputAmperage());
```

Comment:
shouldn't this be `distributed * getOutputAmperage()`?
(described issue on discord)